hackage-security/src/Hackage/Security/Trusted/TCB.hs

```diff
@@ -291,8 +301,9 @@ verifyRole' (trusted -> RoleSpec{roleSpecThreshold = KeyThreshold threshold, ..}
       -- was invalid we would already have thrown an error constructing Signed.
       -- (Similarly, if two signatures were made by the same key, the FromJSON
       -- instance for Signatures would have thrown an error.)
-      unless (length (filter isRoleSpecKey sigs) >= fromIntegral threshold) $
-        throwError $ VerificationErrorSignatures targetPath
+      let nSigs = length (filter isRoleSpecKey sigs)
+      unless (nSigs >= fromIntegral threshold) $
+        throwError $ VerificationErrorSignatures targetPath (fromIntegral threshold) (map someKeyId roleSpecKeys) (map (someKeyId . signatureKey) sigs)
```

Comment:
Below you have a definition `signingKeys` for `map (someKeyId . signatureKey) sigs`.
Better to make these two throws of `VerificationErrorSignatures` look-alike.